### ISSUE TITLE
Aggregations dont respect AI search

### ISSUE THREAD

**Issue by reisner** [CityofEdmonton/text_depot#4]
![Fullscreen_2022-09-29__1_28_PM](https://user-images.githubusercontent.com/490216/193124840-1fff475d-24ce-4a8c-9ffb-ed9991ad7190.png)

All parts of the app that use aggregations ignore the `min_score` parameter in the elasticsearch query. This is not noticable in the regular search, because we use `min_score` of 0, so this doesnt filter results in the query. However, when we run an AI search, we use the `min_score` to filter the query, but this does not filter the aggregations. 

* The issues is mentioned [here](https://stackoverflow.com/questions/58948988/min-score-doesn-t-working-with-aggs-using-elasticsearch), though I couldnt get this solution to work.
*

**Comment by reisner** [comment]
An elasticsearch example:

```
GET /coe_td_council_report_files_test/_search
{
  "size": 0,
  "min_score": 2.5,
  "query": {
      "bool": {
          "must": {
              "simple_query_string": {
                  "query": "budget",
                  "fields": ["text"],
                  "default_operator": "and"
              }
          }
      }
  },
  "aggs": {
    "group_by_index": {
        "terms": {
            "field": "_index"
        }
    }
  }
}

```

Note that the total hits respects `min_score`, but the aggregation result does not:

![Cursor_and_Console_-_Dev_Tools_-_Elastic](https://user-images.githubusercontent.com/490216/193125910-16f076a6-fdd1-49ed-b17b-9fa853c43205.png)

**Comment by reisner** [comment]
By replacing this section:

```
  "aggs": {
    "group_by_index": {
        "terms": {
            "field": "_index"
        }
    }
  }
  ```
  
I found [this](https://discuss.elastic.co/t/accessing-score-in-the-filter-of-aggregations/166877)...so, with the following: (note that the `from` must match the`min_score`)
  
```
  "aggs": {
    "group_by_index": {
      "range": {
          "script": {
            "source": "_score"
          },
          "ranges": [
              {
                  "from": 13
              }
            ]
      },
      "aggs": {
        "ind_group": {
          "terms": {
            "field": "_index"
          }
        }
      }
    }
  }
  ```
  
  This makes the results consistent:
  
![Fullscreen_2022-09-29__1_40_PM](https://user-images.githubusercontent.com/490216/193126767-285022ea-2ccd-4fb0-8a40-38472e44ed4c.png)


However, this makes it a little more complicated to modify all of the aggregate queries (we use aggs for the number of hits, as well as each of the tabs in the results). Also, puzzlingly, setting this `from` value to 0 also makes the results consistent....possibly an elasticsearch bug?

**Comment by reisner** [comment]
Note: current elastic version is `8.1.3`

**Comment by reisner** [comment]
Also tried modifying the scoring script to use a ceiling function, but this still included all results for aggs. Not sure why this doesnt happen for regular search:

```
def sim = cosineSimilarity(params.queryVector, 'source_title_vector') + 1;
if (sim > 1.35) { return(sim) } else { return(0) }
```

**Comment by reisner** [comment]
Looks like this issue is a bug in our elasticsearch version. This will be fixed when updating to elastic v`8.5.0`
